Frameworks/Foundation/NSHTTPURLResponse.mm

```diff
@@ -147,29 +147,60 @@ - (id)copyWithZone:(NSZone*)zone {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (BOOL)supportsSecureCoding {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return YES;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (id)initWithCoder:(NSCoder*)decoder {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (void)encodeWithCoder:(NSCoder*)coder {
+    [coder encodeInteger:_statusCode forKey:@"statusCode"];
+    [coder encodeObject:_allHeaderFields forKey:@"allHeaderFields"];
+    [coder encodeObject:_HTTPVersion forKey:@"HTTPVersion"];
+    [super encodeWithCoder:coder];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (void)encodeWithCoder:(NSCoder*)coder {
-    UNIMPLEMENTED();
+- (instancetype)initWithCoder:(NSCoder*)coder {
+    if (self = [super initWithCoder:coder]) {
+        _statusCode = [coder decodeIntegerForKey:@"statusCode"];
+        _allHeaderFields = [coder decodeObjectOfClass:[NSMutableDictionary class] forKey:@"allHeaderFields"];
```

Comment:
(So you will have to deserialize it into a temporary and then copy it.)

---
In reply to: [72313701](https://github.com/Microsoft/WinObjC/pull/673#discussion_r72313701) [](ancestors = 72313701)